src/lib/rest/curl_context.cpp

```diff
@@ -0,0 +1,94 @@
+#include "curl_context.h"
+
+#include <map>
+
+#include <assert.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+
+static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
+static std::map<std::string, struct curl_context> contexts;
+
+struct curl_context
+get_curl_context(const std::string& key)
+{
+  struct curl_context cc = {NULL, NULL};
+  int s = pthread_mutex_lock(&contexts_mutex);
+  if(s!=0)
+  {
+       LM_X(1,("pthread_mutex_lock"));
```

Comment:
We are open to introduce changes in our "way of working" if that improves the way in which the code in implemented. Even Romans may change their habits :)

However, I'd suggest all the Romans being at Rome (currently some ones are in other continents ;) in order to discuss the best way all together. By the moment, let's implement this in the "current Roman" style.